### PR TITLE
replaced klog.Warningf with an error message in client_go

### LINKED ISSUE #94428: client-go prints a klog warning by default in library code which is not allowed
**What happened**:

A code change was merged that used `klog.Warningf` in library code by default (`NewDefaultClientLoadingRules` changed in a way that impacted all code in the world which uses that default, which is pretty common), which is not allowed by convention in Kube because client libraries may be used in UIs that wish to control output (klog is not consistent with other output).  The code change should have had a callback that allowed clients to optionally decide to print, not print to Warningf by default.

https://github.com/kubernetes/kubernetes/pull/78185#issuecomment-685765692

We should also document the convention for other reviewers that logging by default in client libraries is disallowed.

For instance, a callback would have allowed kubectl to print a human readable message `warning: KUBECONFIG does not exist` in general, and in specific cases (like `kubectl config` or login flows where KUBECONFIG not existing is totally reasonable) not print a message.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by Debanitrkl**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
`client-go` has several instances of `klog.Warningf.`
So, turned these warnings into actual errors that are returned by the functions.

#### Which issue(s) this PR fixes:

Fixes #94428

### CODE REVIEW

File: staging/src/k8s.io/client-go/tools/clientcmd/loader.go [hunk 1]
```diff
@@ -218,7 +218,8 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	}
 
 	if rules.WarnIfAllMissing && len(missingList) > 0 && len(kubeconfigs) == 0 {
-		klog.Warningf("Config not found: %s", strings.Join(missingList, ", "))
+		missingString := strings.Join(missingList, ", ")
+		rules.Warner.Warn(fmt.Sprintf("Config not found: %s", missingString))
```
Comment (yliaog) [inline staging/src/k8s.io/client-go/tools/clientcmd/loader.go:222]: it looks that rules does *not* have Warner method

Comment (Debanitrkl) [reply staging/src/k8s.io/client-go/tools/clientcmd/loader.go:222]: Should I add it there and then make the subsequent changes, I guess changes made here https://github.com/kubernetes/kubernetes/pull/95517/commits/e0b4bce7dea64ab67a654cc9085c5ff50cd4b22d were appropriate

Comment (yliaog) [reply staging/src/k8s.io/client-go/tools/clientcmd/loader.go:222]: e0b4bce looks right to me

Comment (Debanitrkl) [reply staging/src/k8s.io/client-go/tools/clientcmd/loader.go:222]: I have opened a new PR for this #104852 please check that

**Comment by Debanitrkl** [comment]
This has been moved to #104852 closing this issue

**Pull request closed without merging**